src/Cli/Microsoft.DotNet.Cli.Utils/MSBuildPropertyParser.cs

```diff
@@ -65,6 +65,21 @@ public static class MSBuildPropertyParser {
             while(TryConsume(out char? c) && c != ';') {
                 currentValue.Append(c);
             }
+            // we're either at the end or 
+            if (AtEnd()) return;
+            // we're just past a semicolon
+            // if semicolon, we need to check if there are any other = in the string (signifying property pairs)
+            if (input.IndexOf('=', currentPos) != -1) {
+                // there are more = in the string, so eject and let a new key/value pair be parsed
+                return;
+            } else {
```

Comment:
As far as I've been able to tell/test - yes.

Reply:
https://github.com/dotnet/msbuild/blob/518c041f4511a6bc23eb40703b69a94ea46c65fd/src/Shared/PropertyParser.cs It looks like there is reasonable parity between the two, though there are some differences, like this will fail anything without = in it. Not sure if this parser will do that.

Reply:
https://github.com/dotnet/msbuild/blob/b44c689ff99fe65fce40a55e362281384e0fae4a/src/Tasks/CreateItem.cs#L81 I don't see where AdditionalMetadata is populated and that would probably have the equivalent logic for '=', maybe we could ping someone on MSBuild if we want to be overly cautious?